### PR TITLE
feat(linux): Start dbus if not running

### LINKED ISSUE #9363: feat(linux): Allow to install/uninstall keyboards if DBus is not started
Currently, when trying to install a keyboard for a different user the installation might fail with `failed to commit changes to dconf: Cannot autolaunch D-Bus without X11 $DISPLAY`.

To reproduce:

```bash
su --login otheruser
km-package-install -p sil_ipa
```

A workaround is to run `export $(dbus-launch)` first which sets the environment variables `DBUS_SESSION_BUS_ADDRESS` and `DBUS_SESSION_BUS_PID`.

It should be possible to detect in `km-package-install`/`km-package-uninstall` if those variables are not set and run `dbus-launch`.

### PR DESCRIPTION
**Pull request by ermshiperete**
This change checks the DBus environment variables when initializing the `keyman_config` module and starts DBus if the variables are not set. This particularly happens when trying to install or uninstall a keyboard for a different user, which will now be possible with this change.

Fixes #9363.

# User Testing

## Preparations

- The tests should be run on these Linux platforms:
  - **GROUP_FOCAL**: Ubuntu 20.04 Focal with Gnome Shell and X11
  - **GROUP_JAMMY_WAYLAND**: Ubuntu 22.04 Jammy with Gnome Shell and Wayland

- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)

- Reboot
- Create a second user `myuser` without admin privileges

## Tests

**TEST_INST**: Can install a keyboard for the second user

- in a terminal window, login as `myuser` by running `su --login myuser`
- as `myuser` run: `km-package-install -p sil_euro_latin --bcp47 de`
- verify that this doesn't output errors
- logout and login as `myuser`
- verify that the "German (EuroLatin (SIL))" keyboard appears in the toolbar 
- verify that you can switch to and type with that keyboard

**TEST_UNINST**: Can uninstall a keyboard for the second user

- logout and login as regular user again
- in a terminal window, login as `myuser` by running `su --login myuser`
- as `myuser` run: `km-package-uninstall sil_euro_latin`
- verify that this doesn't output errors
- logout and login as `myuser`
- verify that the "German (EuroLatin (SIL))" keyboard no longer appears in the toolbar

### CODE REVIEW

File: linux/keyman-config/keyman_config/__init__.py [hunk 1]
```diff
@@ -77,3 +79,26 @@ def add_standard_arguments(parser):
...
+            ('dbus-launch', '--exit-with-session'),
+            stdout=subprocess.PIPE, check=False).stdout
+        lines = stdout.decode('utf-8').splitlines()
+        for line in lines:
+            equal_sign = line.find('=')
+            if equal_sign <= 0:
+                logging.warning('Got unexpected line from dbus-launch: %s', line)
+                continue
+            name = line[:equal_sign]
+            value = line[equal_sign+1:]
```
Comment (darcywong00) [inline linux/keyman-config/keyman_config/__init__.py:100]: Is this empty if `=` is at the end of the line?

**Review by mcdurdin — APPROVED** [review]

**Comment by ermshiperete** [comment]
When installing the packages for this PR on Focal it's no longer possible to type with Keyman. Most often the keyboards don't show up, and if they do typing doesn't work.

**Comment by ermshiperete** [comment]
Things work after rebooting after the installation.

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_FOCAL: Ubuntu 20.04 Focal with Gnome Shell and X11

* **TEST_INST (FAILED):** Tested with the attached PR build on Ubuntu 20.04 Focal Linux OS (x11) (VM) and here is my observation: 1. Installed the PR build Keyman 17.0.279-beta (package version 17.0.279-1~PR-10863-2470.1+focal1). 2. Restarted the system. 3. Created a second user 'myuser1' without admin privileges. 4. Opened terminal window. Changed to second user 'myuser1'. 5. Ran the above-mentioned command and noticed that it is displaying an error message. 

   



* **TEST_UNINST (FAILED):** Tested with the attached PR build on Ubuntu 20.04 Focal Linux OS (x11) (VM) and here is my observation: 1. Restarted the system. 2. Opened terminal window. Changed to second user 'myuser1'. 5. Ran the above-mentioned (uninstall) command and noticed that it is displaying an error message.

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_JAMMY_WAYLAND: Ubuntu 22.04 Jammy with Gnome Shell and Wayland

* **TEST_INST (PASSED):** Tested with the attached PR build (keyman 17.0.279-beta (package version 17.0.279-1~PR-10863-2470.1+jammy1) on Ubuntu 22.04 Jammy with Gnome Shell and Wayland (Virtual Box) and here is my observation:  1. Created a second user 'myuser' without admin privileges. 2. Changed to second user 'myuser'. 3. Run the above mentioned package-install command on the terminal. 4. Verified it doesn't output any errors. 5. logged out then logged in as 'myuser'. 5. Verified that the "German (EuroLatin(SIL))" keyboard is appearing in the toolbar. 6. Verified that switching to German Keyboard and typing with that keyboard is possible.

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_JAMMY_WAYLAND: Ubuntu 22.04 Jammy with Gnome Shell and Wayland

* **TEST_UNINST (PASSED):** 1. Logged in as a regular user again. 2. Changed to Second user 'myuser' in the terminal window. 3. Run the above mentioned package-uninstall command and verified that it doesn't output any errors. 4. Logged out then logged in as 'myuser'. Verified that the "German (EuroLatin(SIL))" keyboard doesn't appear in the toolbar. Seems to be working as expected.

**Comment by ermshiperete** [comment]
GROUP_FOCAL:

- TEST_INST (SKIPPED): error message is unrelated
- TEST_UNINST (SKIPPED): error message is unrelated

**Comment by ermshiperete** [comment]
@bharanidharanj Could you please run the tests again? I had to change quite a bit. Thanks! Note that on Ubuntu 20.04 Focal you can ignore the "Unable to init server: Could not connect: Connection refused" message.

@keymanapp-test-bot retest

**Review by mcdurdin — APPROVED** [review]

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_FOCAL: Ubuntu 20.04 Focal with Gnome Shell and X11

* **TEST_INST (PASSED):** Retested this PR, as per Eberhard's suggestion with the attached PR build (Keyman 17.0.293-beta (package version 17.0.293-1~PR-10863-2581.1+focal1) on Ubuntu 20.04 Focal Linux (X11) and here is my observation : 1. In the terminal window, logged in to myuser. 2. In myuser, run the above-mentioned command line to install sil_euro_latin keyboard. It shows some irrelevant error message on the terminal window. 3. Logged out and logged in as myuser. Verfied that the "German (EuroLatin (SIL))" keyboard appeared in the toolbar. Able to type in that keyboard in the text editor. 

   


* **TEST_UNINST (PASSED):** 1. Logged out and logged in as regular user again. 2. In terminal window, logged in as myuser. 3. As myuser, run the above mentioned command. It shows up some irrelevant error message on the terminal window. 4. Logged out and logged in as myuser. Verified that the "German (EuroLatin (SIL))" keyboard did not appear in the toolbar.

**Comment by bharanidharanj** [comment]
# Test Results

### GROUP_JAMMY_WAYLAND: Ubuntu 22.04 Jammy with Gnome Shell and Wayland

* **TEST_INST (PASSED):** Retested (as per Eberhard's suggestion) with the attached PR build (Keyman 17.0.293-beta (package vesion 17.0.293-1~PR-10863-2581.1+jammy1) on Ubuntu 20.04 Focal Linux (Wayland) and here is my observation : 1. In the terminal window, logged in to myuser. 2. In myuser, run the above mentioned command line to install sil_euro_latin keyboard. It shows up some irrelevant error message on the terminal window. 3. Logged out and logged in as myuser. Verfied that the "German (EuroLatin (SIL))" keyboard appeared in the toolbar. Able to type in that keyboard in the text editor. 

   

* **TEST_UNINST (PASSED):** 1. Logged out and logged in as regular user again. 2. In a terminal window, logged in as myuser. 3. As myuser, run the above mentioned command . It shows up some irrelevant error message on the terminal window. 4. Logged out and logged in as myuser. Verified that the "German (EuroLatin (SIL))" keyboard did not appear in the toolbar.

**Pull request merged**